### ISSUE TITLE
Lint: is_ascii_digit instead of is_digit

### ISSUE THREAD

**Issue by camsteffen** [rust-lang/rust-clippy#6399]
### What it does

Finds usages of `is_digit` that can be replaced with `is_ascii_digit` or `is_ascii_hexdigit`.

### Categories (optional)

- Kind: style

It is simpler since the radix does not need to be specified.

### Drawbacks

None.

### Example

```rust
fn bad(c: char) {
    c.is_digit(10);
    c.is_digit(16);
}
```

Could be written as:

```rust
fn good(c: char) {
    c.is_ascii_digit();
    c.is_ascii_hexdigit();
}
```

**Comment by rajcspsg** [comment]
can I work on this issue?

**Comment by camsteffen** [comment]
@rajcspsg go for it!

**Comment by bengsparks** [comment]
@camsteffen
This issue seems to not have made much progress. 
Can I submit my solution instead?

**Comment by camsteffen** [comment]
I think @rajcspsg is still working on this at #6400.
